### PR TITLE
chore: remove old alerts and configs keys

### PR DESCRIPTION
**Pull request by pkdotson**
### SUMMARY
This pr removes the old deprecated Legacy Alerts Code and Config Keys for the coming 2.0 release.

### TESTING INSTRUCTIONS
User should not be able to use or set old alerts.

### ADDITIONAL INFORMATION

- [x] Has associated issue: https://github.com/apache/superset/issues/18689
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by betodealmeida — COMMENTED** [review]
I love all the red in this PR!

File: superset/views/alerts.py [hunk 1]
```diff
@@ -109,144 +55,3 @@ def log(self, pk: int) -> FlaskResponse:  # pylint: disable=unused-argument
 class AlertView(BaseAlertReportView):
     route_base = "/alert"
     class_permission_name = "ReportSchedule"
```
[the remark this answers is not in the document]
Comment (pkdotson) [reply superset/views/alerts.py:57]: Yes these are used for the dropdown menu alerts and report link

File: superset/views/alerts.py [hunk 2]
```diff
@@ -17,73 +17,19 @@
 """
 DEPRECATION NOTICE: this module is deprecated and will be removed on 2.0.
 """
```
[the remark this answers is not in the document]
Comment (betodealmeida) [reply superset/views/alerts.py:19]: Shall we remove this deprecation notice then? :)

[the remark this answers is not in the document]
Comment (hughhhh) [reply superset/views/alerts.py:19]: +1 remove this

[the remark this answers is not in the document]
Comment (pkdotson) [reply superset/views/alerts.py]: done

**Review by hughhhh — APPROVED** [review]

**Pull request merged**